app/models/form/question.rb

```diff
@@ -75,6 +75,9 @@ def update_answer_link_name(case_log)
                 else
                   case_log[id].blank? ? "Answer" : "Change"
                 end
+    if id == "earnings"
```

Comment:
Same here really. Is there some property of this question that means it gets treated differently that we can support in the JSON DSL?

Reply:
I might need some help with that though since I couldn't really think of a nice way to do it

Reply:
You could add a `depends_on` key to the suffix similar to how it works for pages and sections? So you'd have something like:

```
"scharge": {
                  "check_answer_label": "Service charge",
                  "header": "What is the service charge?",
                  "hint_text": "For example, for cleaning. Households may receive housing benefit or Universal Credit towards the service charge.",
                  "type": "numeric",
                  "min": 0,
                  "step": 1,
                  "width": 5,
                  "prefix": "£",
                  "suffix": [ 
                    { "label": " every week", "depends_on" : [{ "incfreq": "weekly"}]},
                    { "label": " every month", "depends_on" : [{ "incfreq": "monthly"}]}
                  ],
                  "fields-to-add": ["brent", "scharge", "pscharge", "supcharg"],
                  "result-field": "tcharge",
                  "hidden_in_check_answers": true
                },
```